src/Elasticsearch/Metadata/Document/Factory/CatDocumentMetadataFactory.php

```diff
@@ -30,7 +32,7 @@
  */
 final class CatDocumentMetadataFactory implements DocumentMetadataFactoryInterface
 {
-    public function __construct(private readonly Client $client, private readonly ResourceMetadataCollectionFactoryInterface $resourceMetadataFactory, private readonly ?DocumentMetadataFactoryInterface $decorated = null)
+    public function __construct(private readonly Client|ClientInterface $client, private readonly ResourceMetadataCollectionFactoryInterface $resourceMetadataFactory, private readonly ?DocumentMetadataFactoryInterface $decorated = null)
```

Comment:
should we deprecate the old client? how can we align on Elasticsearch updates in the future? The timeline on this is to release this as part of the 3.2 release which will be in september.

Reply:
I would consider letting the versioning for a `api-platform/elasticsearch` match the `elasticsearch/elastichsearch` [versioning](https://github.com/elastic/elasticsearch-php#versioning) even though it means starting the versioning af `7.*`